lib/puppet/settings/base_setting.rb

```diff
@@ -151,7 +161,11 @@ def set_meta(meta)
   end
 
   def deprecated=(deprecation)
-    raise(ArgumentError, "'#{deprecation}' is an unknown setting deprecation state.  Must be either :completely or :allowed_on_commandline") unless [:completely, :allowed_on_commandline].include?(deprecation)
+    unless [:completely, :allowed_on_commandline].include?(deprecation)
+      # ':completely' and ':allowed_on_commandline' are the names of setting values and should not be translated
+      raise ArgumentError, _("'%{deprecation}' is an unknown setting deprecation state.  Must be either :completely or :allowed_on_commandline") %
```

Comment:
I feel like the original message had some words transposed? I'm having a _really_ hard time parsing "`'foo' is an unknown setting deprecation state.`". I _think_ I might be getting what it's trying to say, but it's phrased really awkwardly, and I'm only just starting to be able to parse it after re-reading it 12+ times. "`'foo' is an unknown deprecation state for a setting.`"?

I don't know that fixing this is actually within the scope of the ticket, but it definitely seemed worth mentioning, especially since I can only imagine how difficult it would be for someone to translate given how hard it is for me to parse.